From f238c0468ccff188f1e0f10083a6674bded2a21a Mon Sep 17 00:00:00 2001
From: Guoqing Jiang <guoqing.jiang@suse.com>
Date: Mon, 28 Mar 2022 14:19:33 +0800
Subject: [PATCH] Revert "dmaengine: idxd: fix wq size store permission state"
Patch-mainline: Never, revert stable patch which is added again in sorted section
References: bsn#161
Modified-by-SEL: No


This reverts commit 4ecf25595273203010bc8318c4aee60ad64037ae.

Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/dma/idxd/sysfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma/idxd/sysfs.c b/drivers/dma/idxd/sysfs.c
index ad46b3c648af..b3ab86ced355 100644
--- a/drivers/dma/idxd/sysfs.c
+++ b/drivers/dma/idxd/sysfs.c
@@ -923,7 +923,7 @@ static ssize_t wq_size_store(struct device *dev,
 	if (!test_bit(IDXD_FLAG_CONFIGURABLE, &idxd->flags))
 		return -EPERM;
 
-	if (idxd->state == IDXD_DEV_ENABLED)
+	if (wq->state != IDXD_WQ_DISABLED)
 		return -EPERM;
 
 	if (size + total_claimed_wq_size(idxd) - wq->size > idxd->max_wq_size)
-- 
2.26.2

